.pipelines/SecretManagement-Official.yml

```diff
@@ -188,3 +194,153 @@ extends:
             packagesToPush: $(drop)/Microsoft.PowerShell.SecretManagement.Library.$(version).nupkg
             nuGetFeedType: external
             publishFeedCredentials: PowerShellNuGetOrgPush
+    - stage: PrepForEv2
+      condition: ne(variables['Build.Reason'], 'Schedule')
+      dependsOn: build
+      variables:
+        drop: $(Pipeline.Workspace)/drop_build_main
+        version: $[ stageDependencies.build.main.outputs['package.version'] ]
+      jobs:
+      - job: CopyEv2FilesToArtifact
+        displayName: Copy Ev2 Files To Artifact
+        variables:
+        - name: ob_outputDirectory
+          value: '$(Build.ArtifactStagingDirectory)/ONEBRANCH_ARTIFACT'
+        pool:
+          timeoutInMinutes: 30
+          type: windows
+        steps:  
+        - task: onebranch.pipeline.signing@1
+          displayName: Sign 1st Party Files
+          inputs:
+            command: 'sign'
+            signing_profile: external_distribution
+            files_to_sign: '**\*.ps1'
+            search_root: '$(Build.SourcesDirectory)/EV2Specs/ServiceGroupRoot/Shell'
+        - download: current
+          displayName: Download artifacts
+        - pwsh: |
+            $modulePath = Join-Path $(drop) -ChildPath 'Microsoft.PowerShell.SecretManagement.$(version).nupkg'
+            $fileExists = Test-Path $modulePath
+            Write-Verbose -Verbose "Module file $modulePath Exists: $fileExists"
+
+            $srcDir = Join-Path '$(Build.SourcesDirectory)/EV2Specs/ServiceGroupRoot' -ChildPath 'SrcFiles'
+            New-Item $srcDir -ItemType Directory
+            
+            $dest = Join-Path $srcDir -ChildPath 'Microsoft.PowerShell.SecretManagement.$(version).nupkg'
+            Copy-Item -Path $modulePath -Destination $dest
+            
+            Get-ChildItem '$(Build.SourcesDirectory)/EV2Specs/ServiceGroupRoot/SrcFiles/'
+          displayName: Move artifact to Ev2 folder
+        - pwsh: |
+            $srcPath = Join-Path '$(Build.SourcesDirectory)/EV2Specs/ServiceGroupRoot' -ChildPath 'Shell'
+            $pathToRunTarFile = Join-Path '$(Build.SourcesDirectory)/EV2Specs/ServiceGroupRoot/Shell' -ChildPath "Run.tar"
+            tar -cvf $pathToRunTarFile -C $srcPath ./Run
+            $tarExists = Test-Path $pathToRunTarFile
+            Write-Verbose -Verbose "Tar file $pathToRunTarFile exists: $tarExists"
+          displayName: Compress Run script into tar file as needed for EV2 Shell extension           
+        - pwsh: |
+            $pathToJsonFile = Join-Path -Path '$(Build.SourcesDirectory)/EV2Specs/ServiceGroupRoot' -ChildPath 'SecretManagementToACR.Rollout.json'
+            $content = Get-Content -Path $pathToJsonFile | ConvertFrom-Json
+            $envVarArrayLen = $content.shellExtensions.launch.environmentVariables.Length
+
+            [xml]$xml = Get-Content Directory.Build.props
+            $version = $xml.Project.PropertyGroup.ModuleVersion
+
+            for ($i=0; $i -lt $envVarArrayLen; $i++)
```

Comment:
Per Viv's idea: we remove the predefined values from the `environmentValues` array and simply build its contents here, instead of having to iterate over it with a bunch of checks and sets.